beard/utils/names.py

```diff
@@ -46,3 +50,69 @@ def normalize_name(name):
 def name_initials(name):
     """Compute the set of initials of a given name."""
     return set([w[0] for w in name.split()])
+
+RE_APOSTROPHES = re.compile('\'+')
+RE_REMOVE_NON_CHARACTERS = re.compile('[^a-zA-Z\',\s]+')
+DROPPED_PREFIXES = {'a', 'ab', 'am', 'ap', 'abu', 'al', 'auf', 'aus', 'bar',
+                    'bath', 'bat', 'ben', 'bet', 'bin', 'bint', 'd', 'da',
+                    'dall', 'dalla', 'das', 'de', 'degli', 'del', 'dell',
+                    'della', 'dem', 'den', 'der', 'di', 'do', 'dos', 'du', 'e',
+                    'el', 'i', 'ibn', 'im', 'jr', 'l', 'la', 'las', 'le',
+                    'los', 'm', 'mac', 'mc', 'mhic', 'mic', 'o', 'ter', 'und',
+                    'v', 'van', 'vom', 'von', 'zu', 'zum', 'zur'}
+
+
+@memoize
+def tokenize_name(name):
+    """Create Double Metaphone tokens from the string.
+
+    Preprocess the name.
+
+     Parameters
+    ----------
+    :param name: name of the author
+
+    Returns
+    -------
+    :return tuple: the tokens in form (tuple of strings, tuple of strings)
+        representing last names and first names
+    """
+    name = asciify(name)
+
+    # Get rid of non character. Leave apostrophes as they are handled in a
+    # different way.
+    name = RE_REMOVE_NON_CHARACTERS.sub(' ', name)
+
+    # Handle the "miagkii znak" in russian names.
+    matches = re.findall(r"^([^',]*)'([a-z].*)", name)
+    if matches:
+        name = matches[0][0] + matches[0][1]
+
+    # Remove apostrophes
+    name = RE_APOSTROPHES.sub(' ', name)
+
+    # Extract surname and name
+    tokens = name.split(',')
```

Comment:
Your function should work, but this makes me think we should maybe update the docstring to describe the input format of names.